### ISSUE TITLE
Username disappears after a period of time from the context

### ISSUE THREAD

**Issue by andrewyager** [realworldtech/radius_restserver#24]
After a period of time the username drops out of the context.

**Comment by andrewyager** [comment]
@DanielGabris can you look at this next? The username test@testing.com / testing123 should log you into the application with the test data loaded.

**Comment by DanielGabris** [comment]
@andrewyager Okay, I will do it. Thanks

**Comment by DanielGabris** [comment]
@andrewyager I am still getting django module import error for django-radius.
I have installed it successfully through pip install and all packages in requirement.txt are installed.
But it is not finding the module radiusauth.backend in settings.py.
What do you think the problem is? Which django version are you using?

**Comment by andrewyager** [comment]
@DanielGabris at what stage are you getting this error? All the requirements are installed automatically as part of the docker build for the django image.

Just to be sure, I've just reset my docker machine, have cloned a clean repository and am now running a build to test and will provide the build output so you can compare it against yours.

**Comment by DanielGabris** [comment]
@andrewyager Okay, maybe I will need to rebuild docker. I will let you know how it work.

**Comment by andrewyager** [comment]
@DanielGabris i've just finished building and running the test suite on my local machine and it does work.

```
Last login: Mon Feb  1 16:13:14 on ttys000
**my-machine:~ andrewya$ docker-machine start default
Started machines may have new IP addresses. You may need to re-run the `docker-machine env` command.
**my-machine:~ andrewya$ eval "$(docker-machine env default)"
**my-machine:~ andrewya$ git clone https://github.com/realworldtech/radius_restserver testclone
Cloning into 'testclone'...
remote: Counting objects: 634, done.
remote: Compressing objects: 100% (71/71), done.
remote: Total 634 (delta 25), reused 0 (delta 0), pack-reused 562
Receiving objects: 100% (634/634), 219.34 KiB | 266.00 KiB/s, done.
Resolving deltas: 100% (312/312), done.
Checking connectivity... done.
**my-machine:~ andrewya$ cd testclone/
**my-machine:testclone andrewya$ docker-compose build
mysqldb uses an image, skipping
radius uses an image, skipping
dbadmin uses an image, skipping
postgres uses an image, skipping
Building djangoproduction...
Step 1 : FROM django:1.8-python2
Pulling repository docker.io/library/django
ed86419ecd18: Download complete
5679b9b90e09: Download complete
ea6bab360f56: Download complete
696e9c5bf8c6: Download complete
0e803dbc6621: Download complete
d5f544521b95: Download complete
72e72455b4be: Download complete
722882d9ac92: Download complete
1a6ab8ef5c38: Download complete
804f466cb234: Download complete
a2f7015ad312: Download complete
daea18aca18f: Download complete
88f2d57f7209: Download complete
35fff7fcbe84: Download complete
Status: Downloaded newer image for django:1.8-python2
docker.io/library/django: this image was pulled from a legacy registry.  Important: This registry version will not be supported in future versions of docker.
 ---> ed86419ecd18
Step 2 : RUN mkdir -p /usr/src/app
 ---> Running in 781d0de17b3c
 ---> d656dfa97d18
Removing intermediate container 781d0de17b3c
Step 3 : WORKDIR /usr/src/app
 ---> Running in 2d0028ff8bca
 ---> 99f86da4b0d6
Removing intermediate container 2d0028ff8bca
Step 4 : RUN apt-get update
 ---> Running in 047eb9caa51e
Get:1 http://security.debian.org jessie/updates InRelease [63.1 kB]
Get:2 http://security.debian.org jessie/updates/main amd64 Packages [252 kB]
Ign http://httpredir.debian.org jessie InRelease
Get:3 http://httpredir.debian.org jessie-updates InRelease [136 kB]
Get:4 http://httpredir.debian.org jessie Release.gpg [2373 B]
Get:5 http://httpredir.debian.org jessie-updates/main amd64 Packages [3619 B]
Get:6 http://httpredir.debian.org jessie Release [148 kB]
Get:7 http://httpredir.debian.org jessie/main amd64 Packages [9034 kB]
Fetched 9639 kB in 9s (1007 kB/s)
Reading package lists...
 ---> c5a9b35e7268
Removing intermediate container 047eb9caa51e
Step 5 : RUN apt-get install -y 		mysql-client libmysqlclient-dev 		postgresql-client libpq-dev 		sqlite3 		git-core 		gcc
 ---> Running in 3ec69c736145
Reading package lists...
Building dependency tree...
Reading state information...
gcc is already the newest version.
libpq-dev is already the newest version.
postgresql-client is already the newest version.
sqlite3 is already the newest version.
The following extra packages will be installed:
  git git-man less libcurl3-gnutls liberror-perl libexpat1 libidn11
  libmysqlclient18 libpopt0 librtmp1 libssh2-1 libx11-6 libx11-data libxau6
  libxcb1 libxdmcp6 libxext6 libxmuu1 mysql-common openssh-client patch rsync
  xauth
Suggested packages:
  git-daemon-run git-daemon-sysvinit git-doc git-el git-email git-gui gitk
  gitweb git-arch git-cvs git-mediawiki git-svn ssh-askpass libpam-ssh
  keychain monkeysphere ed diffutils-doc openssh-server
Recommended packages:
  ssh-client
The following NEW packages will be installed:
  git git-core git-man less libcurl3-gnutls liberror-perl libexpat1 libidn11
  libpopt0 librtmp1 libssh2-1 libx11-6 libx11-data libxau6 libxcb1 libxdmcp6
  libxext6 libxmuu1 openssh-client patch rsync xauth
The following packages will be upgraded:
  libmysqlclient-dev libmysqlclient18 mysql-client mysql-common
4 upgraded, 22 newly installed, 0 to remove and 32 not upgraded.
Need to get 9419 kB of archives.
After this operation, 34.1 MB of additional disk space will be used.
Get:1 http://security.debian.org/ jessie/updates/main libcurl3-gnutls amd64 7.38.0-4+deb8u3 [251 kB]
Get:2 http://security.debian.org/ jessie/updates/main libmysqlclient-dev amd64 5.5.47-0+deb8u1 [957 kB]
Get:3 http://security.debian.org/ jessie/updates/main mysql-common all 5.5.47-0+deb8u1 [82.6 kB]
Get:4 http://security.debian.org/ jessie/updates/main libmysqlclient18 amd64 5.5.47-0+deb8u1 [675 kB]
Get:5 http://httpredir.debian.org/debian/ jessie/main libpopt0 amd64 1.16-10 [49.2 kB]
Get:6 http://security.debian.org/ jessie/updates/main mysql-client all 5.5.47-0+deb8u1 [80.7 kB]
Get:7 http://httpredir.debian.org/debian/ jessie/main librtmp1 amd64 2.4+20150115.gita107cef-1 [59.8 kB]
Get:8 http://httpredir.debian.org/debian/ jessie/main libidn11 amd64 1.29-1+b2 [136 kB]
Get:9 http://httpredir.debian.org/debian/ jessie/main libssh2-1 amd64 1.4.3-4.1 [125 kB]
Get:10 http://httpredir.debian.org/debian/ jessie/main libx11-data all 2:1.6.2-3 [126 kB]
Get:11 http://httpredir.debian.org/debian/ jessie/main libexpat1 amd64 2.1.0-6+deb8u1 [80.0 kB]
Get:12 http://httpredir.debian.org/debian/ jessie/main libxau6 amd64 1:1.0.8-1 [20.7 kB]
Get:13 http://httpredir.debian.org/debian/ jessie/main libx11-6 amd64 2:1.6.2-3 [729 kB]
Get:14 http://httpredir.debian.org/debian/ jessie/main openssh-client amd64 1:6.7p1-5+deb8u1 [691 kB]
Get:15 http://httpredir.debian.org/debian/ jessie/main git amd64 1:2.1.4-2.1+deb8u1 [3258 kB]
Get:16 http://httpredir.debian.org/debian/ jessie/main libxdmcp6 amd64 1:1.1.1-1+b1 [24.9 kB]
Get:17 http://httpredir.debian.org/debian/ jessie/main patch amd64 2.7.5-1 [109 kB]
Get:18 http://httpredir.debian.org/debian/ jessie/main liberror-perl all 0.17-1.1 [22.4 kB]
Get:19 http://httpredir.debian.org/debian/ jessie/main rsync amd64 3.1.1-3 [390 kB]
Get:20 http://httpredir.debian.org/debian/ jessie/main libxcb1 amd64 1.10-3+b1 [44.4 kB]
Get:21 http://httpredir.debian.org/debian/ jessie/main xauth amd64 1:1.0.9-1 [38.2 kB]
Get:22 http://httpredir.debian.org/debian/ jessie/main less amd64 458-3 [124 kB]
Get:23 http://httpredir.debian.org/debian/ jessie/main git-man all 1:2.1.4-2.1+deb8u1 [1267 kB]
Get:24 http://httpredir.debian.org/debian/ jessie/main libxext6 amd64 2:1.3.3-1 [52.7 kB]
Get:25 http://httpredir.debian.org/debian/ jessie/main libxmuu1 amd64 2:1.1.2-1 [23.3 kB]
Get:26 http://httpredir.debian.org/debian/ jessie/main git-core all 1:2.1.4-2.1+deb8u1 [1486 B]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 9419 kB in 32s (294 kB/s)
Selecting previously unselected package libpopt0:amd64.
(Reading database ... 12316 files and directories currently installed.)
Preparing to unpack .../libpopt0_1.16-10_amd64.deb ...
Unpacking libpopt0:amd64 (1.16-10) ...
Selecting previously unselected package libidn11:amd64.
Preparing to unpack .../libidn11_1.29-1+b2_amd64.deb ...
Unpacking libidn11:amd64 (1.29-1+b2) ...
Selecting previously unselected package librtmp1:amd64.
Preparing to unpack .../librtmp1_2.4+20150115.gita107cef-1_amd64.deb ...
Unpacking librtmp1:amd64 (2.4+20150115.gita107cef-1) ...
Selecting previously unselected package libssh2-1:amd64.
Preparing to unpack .../libssh2-1_1.4.3-4.1_amd64.deb ...
Unpacking libssh2-1:amd64 (1.4.3-4.1) ...
Selecting previously unselected package libcurl3-gnutls:amd64.
Preparing to unpack .../libcurl3-gnutls_7.38.0-4+deb8u3_amd64.deb ...
Unpacking libcurl3-gnutls:amd64 (7.38.0-4+deb8u3) ...
Selecting previously unselected package libexpat1:amd64.
Preparing to unpack .../libexpat1_2.1.0-6+deb8u1_amd64.deb ...
Unpacking libexpat1:amd64 (2.1.0-6+deb8u1) ...
Preparing to unpack .../libmysqlclient-dev_5.5.47-0+deb8u1_amd64.deb ...
Unpacking libmysqlclient-dev (5.5.47-0+deb8u1) over (5.5.46-0+deb8u1) ...
Preparing to unpack .../mysql-common_5.5.47-0+deb8u1_all.deb ...
Unpacking mysql-common (5.5.47-0+deb8u1) over (5.5.46-0+deb8u1) ...
Preparing to unpack .../libmysqlclient18_5.5.47-0+deb8u1_amd64.deb ...
Unpacking libmysqlclient18:amd64 (5.5.47-0+deb8u1) over (5.5.46-0+deb8u1) ...
Selecting previously unselected package libxau6:amd64.
Preparing to unpack .../libxau6_1%3a1.0.8-1_amd64.deb ...
Unpacking libxau6:amd64 (1:1.0.8-1) ...
Selecting previously unselected package libxdmcp6:amd64.
Preparing to unpack .../libxdmcp6_1%3a1.1.1-1+b1_amd64.deb ...
Unpacking libxdmcp6:amd64 (1:1.1.1-1+b1) ...
Selecting previously unselected package libxcb1:amd64.
Preparing to unpack .../libxcb1_1.10-3+b1_amd64.deb ...
Unpacking libxcb1:amd64 (1.10-3+b1) ...
Selecting previously unselected package libx11-data.
Preparing to unpack .../libx11-data_2%3a1.6.2-3_all.deb ...
Unpacking libx11-data (2:1.6.2-3) ...
Selecting previously unselected package libx11-6:amd64.
Preparing to unpack .../libx11-6_2%3a1.6.2-3_amd64.deb ...
Unpacking libx11-6:amd64 (2:1.6.2-3) ...
Selecting previously unselected package libxext6:amd64.
Preparing to unpack .../libxext6_2%3a1.3.3-1_amd64.deb ...
Unpacking libxext6:amd64 (2:1.3.3-1) ...
Selecting previously unselected package libxmuu1:amd64.
Preparing to unpack .../libxmuu1_2%3a1.1.2-1_amd64.deb ...
Unpacking libxmuu1:amd64 (2:1.1.2-1) ...
Selecting previously unselected package less.
Preparing to unpack .../archives/less_458-3_amd64.deb ...
Unpacking less (458-3) ...
Selecting previously unselected package openssh-client.
Preparing to unpack .../openssh-client_1%3a6.7p1-5+deb8u1_amd64.deb ...
Unpacking openssh-client (1:6.7p1-5+deb8u1) ...
Selecting previously unselected package patch.
Preparing to unpack .../patch_2.7.5-1_amd64.deb ...
Unpacking patch (2.7.5-1) ...
Selecting previously unselected package liberror-perl.
Preparing to unpack .../liberror-perl_0.17-1.1_all.deb ...
Unpacking liberror-perl (0.17-1.1) ...
Selecting previously unselected package git-man.
Preparing to unpack .../git-man_1%3a2.1.4-2.1+deb8u1_all.deb ...
Unpacking git-man (1:2.1.4-2.1+deb8u1) ...
Selecting previously unselected package git.
Preparing to unpack .../git_1%3a2.1.4-2.1+deb8u1_amd64.deb ...
Unpacking git (1:2.1.4-2.1+deb8u1) ...
Selecting previously unselected package git-core.
Preparing to unpack .../git-core_1%3a2.1.4-2.1+deb8u1_all.deb ...
Unpacking git-core (1:2.1.4-2.1+deb8u1) ...
Preparing to unpack .../mysql-client_5.5.47-0+deb8u1_all.deb ...
Unpacking mysql-client (5.5.47-0+deb8u1) over (5.5.46-0+deb8u1) ...
Selecting previously unselected package rsync.
Preparing to unpack .../rsync_3.1.1-3_amd64.deb ...
Unpacking rsync (3.1.1-3) ...
Selecting previously unselected package xauth.
Preparing to unpack .../xauth_1%3a1.0.9-1_amd64.deb ...
Unpacking xauth (1:1.0.9-1) ...
Processing triggers for systemd (215-17+deb8u2) ...
Setting up libpopt0:amd64 (1.16-10) ...
Setting up libidn11:amd64 (1.29-1+b2) ...
Setting up librtmp1:amd64 (2.4+20150115.gita107cef-1) ...
Setting up libssh2-1:amd64 (1.4.3-4.1) ...
Setting up libcurl3-gnutls:amd64 (7.38.0-4+deb8u3) ...
Setting up libexpat1:amd64 (2.1.0-6+deb8u1) ...
Setting up mysql-common (5.5.47-0+deb8u1) ...
Setting up libmysqlclient18:amd64 (5.5.47-0+deb8u1) ...
Setting up libmysqlclient-dev (5.5.47-0+deb8u1) ...
Setting up libxau6:amd64 (1:1.0.8-1) ...
Setting up libxdmcp6:amd64 (1:1.1.1-1+b1) ...
Setting up libxcb1:amd64 (1.10-3+b1) ...
Setting up libx11-data (2:1.6.2-3) ...
Setting up libx11-6:amd64 (2:1.6.2-3) ...
Setting up libxext6:amd64 (2:1.3.3-1) ...
Setting up libxmuu1:amd64 (2:1.1.2-1) ...
Setting up less (458-3) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
Setting up openssh-client (1:6.7p1-5+deb8u1) ...
Setting up patch (2.7.5-1) ...
Setting up liberror-perl (0.17-1.1) ...
Setting up git-man (1:2.1.4-2.1+deb8u1) ...
Setting up git (1:2.1.4-2.1+deb8u1) ...
Setting up git-core (1:2.1.4-2.1+deb8u1) ...
Setting up mysql-client (5.5.47-0+deb8u1) ...
Setting up rsync (3.1.1-3) ...
invoke-rc.d: policy-rc.d denied execution of restart.
Setting up xauth (1:1.0.9-1) ...
Processing triggers for libc-bin (2.19-18+deb8u1) ...
Processing triggers for systemd (215-17+deb8u2) ...
 ---> c35ed7be0ddb
Removing intermediate container 3ec69c736145
Step 6 : RUN apt-get install -y freeradius-common
 ---> Running in 36f53c08827b
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  freeradius-common
0 upgraded, 1 newly installed, 0 to remove and 32 not upgraded.
Need to get 227 kB of archives.
After this operation, 950 kB of additional disk space will be used.
Get:1 http://httpredir.debian.org/debian/ jessie/main freeradius-common all 2.2.5+dfsg-0.2 [227 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 227 kB in 0s (237 kB/s)
Selecting previously unselected package freeradius-common.
(Reading database ... 13585 files and directories currently installed.)
Preparing to unpack .../freeradius-common_2.2.5+dfsg-0.2_all.deb ...
Unpacking freeradius-common (2.2.5+dfsg-0.2) ...
Setting up freeradius-common (2.2.5+dfsg-0.2) ...
Adding user freerad to group shadow
 ---> 8462e252f41f
Removing intermediate container 36f53c08827b
Step 7 : COPY requirements.txt /usr/src/app/
 ---> 02e58f520af2
Removing intermediate container 1e03a0bc683e
Step 8 : RUN pip install -r /usr/src/app/requirements.txt
 ---> Running in 4646c39d2b26
Collecting pyrad==2.0 (from -r /usr/src/app/requirements.txt (line 1))
  Downloading pyrad-2.0.tar.gz
Obtaining django-radius from git+git://github.com/andrewyager/django-radius.git#egg=django-radius (from -r /usr/src/app/requirements.txt (line 2))
  Cloning git://github.com/andrewyager/django-radius.git to ./src/django-radius
Collecting django-filter==0.11.0 (from -r /usr/src/app/requirements.txt (line 3))
  Downloading django_filter-0.11.0-py2.py3-none-any.whl
Collecting djangorestframework==3.3.1 (from -r /usr/src/app/requirements.txt (line 4))
  Downloading djangorestframework-3.3.1-py2.py3-none-any.whl (576kB)
Collecting Markdown==2.6.4 (from -r /usr/src/app/requirements.txt (line 5))
  Downloading Markdown-2.6.4.tar.gz (301kB)
Collecting django-admin-bootstrapped==2.5.6 (from -r /usr/src/app/requirements.txt (line 6))
  Downloading django-admin-bootstrapped-2.5.6.tar.gz (378kB)
Collecting django-bootstrap3==6.2.2 (from -r /usr/src/app/requirements.txt (line 7))
  Downloading django-bootstrap3-6.2.2.tar.gz
Collecting django-cors-headers==1.1.0 (from -r /usr/src/app/requirements.txt (line 8))
  Downloading django-cors-headers-1.1.0.tar.gz
Collecting djangorestframework-jwt==1.7.2 (from -r /usr/src/app/requirements.txt (line 9))
  Downloading djangorestframework_jwt-1.7.2-py2.py3-none-any.whl
Collecting django-rest-auth (from -r /usr/src/app/requirements.txt (line 10))
  Downloading django-rest-auth-0.6.0.tar.gz
Collecting django-configurations (from -r /usr/src/app/requirements.txt (line 11))
  Downloading django_configurations-1.0-py2.py3-none-any.whl
Collecting python-dateutil (from -r /usr/src/app/requirements.txt (line 12))
  Downloading python_dateutil-2.4.2-py2.py3-none-any.whl (188kB)
Collecting six (from pyrad==2.0->-r /usr/src/app/requirements.txt (line 1))
  Downloading six-1.10.0-py2.py3-none-any.whl
Requirement already satisfied (use --upgrade to upgrade): setuptools in /usr/local/lib/python2.7/site-packages (from django-admin-bootstrapped==2.5.6->-r /usr/src/app/requirements.txt (line 6))
Requirement already satisfied (use --upgrade to upgrade): Django>=1.8 in /usr/local/lib/python2.7/site-packages (from django-admin-bootstrapped==2.5.6->-r /usr/src/app/requirements.txt (line 6))
Collecting PyJWT<2.0.0,>=1.4.0 (from djangorestframework-jwt==1.7.2->-r /usr/src/app/requirements.txt (line 9))
  Downloading PyJWT-1.4.0-py2.py3-none-any.whl
Building wheels for collected packages: pyrad, Markdown, django-admin-bootstrapped, django-bootstrap3, django-cors-headers, django-rest-auth
  Running setup.py bdist_wheel for pyrad
  Stored in directory: /root/.cache/pip/wheels/43/04/d6/4c9e66539baf6c6f006f48ef830bb2c259330237b5f3667406
  Running setup.py bdist_wheel for Markdown
  Stored in directory: /root/.cache/pip/wheels/60/74/5f/530b6b568597290fc5a3caa3ab83b7305dc3cdc3b9c3a7da70
  Running setup.py bdist_wheel for django-admin-bootstrapped
  Stored in directory: /root/.cache/pip/wheels/87/34/c4/1fb802cc737f7ab1f3c44e5920024970b1e935dfb4ddd1085a
  Running setup.py bdist_wheel for django-bootstrap3
  Stored in directory: /root/.cache/pip/wheels/fe/58/ff/c909935dd8ed28866f71a850f998dc350fbadd2a3b942abe62
  Running setup.py bdist_wheel for django-cors-headers
  Stored in directory: /root/.cache/pip/wheels/e5/03/cc/e61e063db325f0bae38dc5e2f390fbaca4a5497c1dd63d49ab
  Running setup.py bdist_wheel for django-rest-auth
  Stored in directory: /root/.cache/pip/wheels/af/c7/1a/158a2e4ea5e9491670b63a3c6f981c6aed00cd7d8f4327973f
Successfully built pyrad Markdown django-admin-bootstrapped django-bootstrap3 django-cors-headers django-rest-auth
Installing collected packages: six, pyrad, django-radius, django-filter, djangorestframework, Markdown, django-admin-bootstrapped, django-bootstrap3, django-cors-headers, PyJWT, djangorestframework-jwt, django-rest-auth, django-configurations, python-dateutil
  Running setup.py develop for django-radius
Successfully installed Markdown-2.6.4 PyJWT-1.4.0 django-admin-bootstrapped-2.5.6 django-bootstrap3-6.2.2 django-configurations-1.0 django-cors-headers-1.1.0 django-filter-0.11.0 django-radius django-rest-auth-0.6.0 djangorestframework-3.3.1 djangorestframework-jwt-1.7.2 pyrad-2.0 python-dateutil-2.4.2 six-1.10.0
You are using pip version 7.1.2, however version 8.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
 ---> 9e75ef74ce80
Removing intermediate container 4646c39d2b26
Step 9 : EXPOSE 8000
 ---> Running in 5b2e553bb52c
 ---> 0fdbfb3683f3
Removing intermediate container 5b2e553bb52c
Step 10 : CMD python manage.py runserver 0.0.0.0:8000
 ---> Running in 2dc03e90b928
 ---> 81d8e378ae86
Removing intermediate container 2dc03e90b928
Successfully built 81d8e378ae86
Building frontend...
Step 1 : FROM digitallyseamless/nodejs-bower-grunt
 ---> 7b37adb8661e
Step 2 : CMD grunt serve
 ---> Using cache
 ---> 419c64dc5cbf
Successfully built 419c64dc5cbf
Building django...
Step 1 : FROM django:1.8-python2
 ---> ed86419ecd18
Step 2 : RUN mkdir -p /usr/src/app
 ---> Using cache
 ---> d656dfa97d18
Step 3 : WORKDIR /usr/src/app
 ---> Using cache
 ---> 99f86da4b0d6
Step 4 : RUN apt-get update
 ---> Using cache
 ---> c5a9b35e7268
Step 5 : RUN apt-get install -y 		mysql-client libmysqlclient-dev 		postgresql-client libpq-dev 		sqlite3 		git-core 		gcc
 ---> Using cache
 ---> c35ed7be0ddb
Step 6 : RUN apt-get install -y freeradius-common
 ---> Using cache
 ---> 8462e252f41f
Step 7 : COPY requirements.txt /usr/src/app/
 ---> Using cache
 ---> 02e58f520af2
Step 8 : RUN pip install -r /usr/src/app/requirements.txt
 ---> Using cache
 ---> 9e75ef74ce80
Step 9 : EXPOSE 8000
 ---> Using cache
 ---> 0fdbfb3683f3
Step 10 : CMD python manage.py runserver 0.0.0.0:8000
 ---> Using cache
 ---> 81d8e378ae86
Successfully built 81d8e378ae86
**my-machine:testclone andrewya$ docker-compose up -d
Creating testclone_mysqldb_1...
Pulling radius (realworld/radius:latest)...
latest: Pulling from realworld/radius
40ce9e2a3945: Pull complete
ca8454887ef8: Pull complete
0fb6893508dc: Pull complete
82a3577ae081: Pull complete
c86db186c84e: Pull complete
1460575f3209: Pull complete
c0cb01f07836: Pull complete
dde8b00c727b: Pull complete
8d6112567da6: Pull complete
87b6a67501d5: Pull complete
33f24cd16372: Pull complete
9d68a4b6cdb0: Pull complete
89e1447686cd: Pull complete
02f2beceef3e: Pull complete
b7cad5cf8103: Pull complete
ee3c8de4ce0c: Pull complete
aceabf300bf5: Pull complete
410274c93076: Pull complete
afd06c3424c4: Pull complete
772a426c7934: Pull complete
ee4bc7a67bfd: Pull complete
4482e8998f9c: Pull complete
f28456969082: Pull complete
a1b65333bb07: Pull complete
Digest: sha256:7ceaf1514c660ea86ee8fb9e8234fe1e5aa742a63dd152054651abb8d48b2afc
Status: Downloaded newer image for realworld/radius:latest
Creating testclone_radius_1...
Pulling dbadmin (marvambass/phpmyadmin:latest)...
latest: Pulling from marvambass/phpmyadmin
d634beec75db: Pull complete
e32ab2fa6818: Pull complete
b60cb1b3cff1: Pull complete
abd23a5068b8: Pull complete
1ec19dfe439e: Pull complete
603cfe879c0c: Pull complete
e027e9274e90: Pull complete
fc4384eeda30: Pull complete
7d62b6b63b7f: Pull complete
651bdcc5b0b4: Pull complete
f3ae5aa055ff: Pull complete
d6593cdf6a0e: Pull complete
1da96d8e4ca6: Pull complete
bc7d857cc8c0: Pull complete
c0992d61eefb: Pull complete
4f773e3fa55a: Pull complete
9b56f29b6a7b: Pull complete
85ae96eba526: Pull complete
3cbf3b03745f: Pull complete
b3c0249a6425: Pull complete
40d85990772f: Pull complete
f5d76aa5eb76: Pull complete
81af3dacfd3a: Pull complete
43dbfc005c52: Pull complete
3ab65e9a99a7: Pull complete
1f6136d848e7: Pull complete
83334b1a1fbe: Pull complete
1890ff0f3560: Pull complete
eeb254e56758: Pull complete
78a98928ffb0: Pull complete
05d7b70c8fc1: Pull complete
3d6d84ac6916: Pull complete
fe3e38ae43ff: Pull complete
40c278071763: Pull complete
b280efdf04db: Pull complete
387013c120e3: Pull complete
70ae7f0dc4ee: Pull complete
8f9f92aeb13a: Pull complete
a4f6c6b8b595: Pull complete
221bb426f891: Pull complete
a49eb2f44398: Pull complete
dcf0871b381c: Pull complete
43fc8f124712: Pull complete
9dbf5870b957: Pull complete
Digest: sha256:b332e9a53ee3b8a30fe6347652eea47d2845bfcc671f09e1aa28c7020246ebe7
Status: Downloaded newer image for marvambass/phpmyadmin:latest
Creating testclone_dbadmin_1...
Pulling postgres (postgres:9.4)...
9.4: Pulling from library/postgres
822bd1a2442c: Pull complete
5073b8150307: Pull complete
720f774b0090: Pull complete
b0c424cc1f71: Pull complete
8e2afdde2f0c: Pull complete
9f5f7dfc92ed: Pull complete
62134ff76329: Pull complete
9d9c96fb6a64: Pull complete
2e4a24d5467c: Pull complete
b0db98630d9c: Pull complete
2bf5d57b8133: Pull complete
d7456fe9fb67: Pull complete
ef0e43aab7a5: Pull complete
b964cff523bc: Pull complete
44196bd473a2: Pull complete
bd77202e4891: Pull complete
92ca55213e25: Pull complete
6b1fe5978300: Pull complete
de35bfc3a186: Pull complete
Digest: sha256:47e1973f3cc987442613ae8f274d1a4861133f291202473fc15a5a012582f3bd
Status: Downloaded newer image for postgres:9.4
Creating testclone_postgres_1...
Creating testclone_djangoproduction_1...
Creating testclone_frontend_1...
Creating testclone_django_1...
**my-machine:testclone andrewya$ docker-compose ps
            Name                          Command                State                Ports             
-------------------------------------------------------------------------------------------------------
testclone_dbadmin_1            /opt/entrypoint.sh nginx         Up        0.0.0.0:8443->443/tcp, 80/tcp 
testclone_django_1             python manage.py runserver ...   Up        0.0.0.0:81->8000/tcp          
testclone_djangoproduction_1   python manage.py runserver ...   Exit 1                                  
testclone_frontend_1           grunt serve                      Exit 99                                 
testclone_mysqldb_1            /entrypoint.sh mysqld            Up        3306/tcp                      
testclone_postgres_1           /docker-entrypoint.sh postgres   Up        5432/tcp                      
testclone_radius_1             /bin/sh -c /bin/bash /usr/ ...   Up        1812/udp, 1813/udp            
**my-machine:testclone andrewya$ docker-compose logs django
Attaching to testclone_django_1
django_1 | django-configurations version 1.0, using configuration 'Dev'
django_1 | Unhandled exception in thread started by <function wrapper at 0x7f3b6ed46b18>
django_1 | Performing system checks...
django_1 | 
django_1 | System check identified no issues (0 silenced).
django_1 | Traceback (most recent call last):
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/utils/autoreload.py", line 229, in wrapper
django_1 |     fn(*args, **kwargs)
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 116, in inner_run
django_1 |     self.check_migrations()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 168, in check_migrations
django_1 |     executor = MigrationExecutor(connections[DEFAULT_DB_ALIAS])
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/executor.py", line 19, in __init__
django_1 |     self.loader = MigrationLoader(self.connection)
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/loader.py", line 47, in __init__
django_1 |     self.build_graph()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/loader.py", line 191, in build_graph
django_1 |     self.applied_migrations = recorder.applied_migrations()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/recorder.py", line 59, in applied_migrations
django_1 |     self.ensure_schema()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/recorder.py", line 49, in ensure_schema
django_1 |     if self.Migration._meta.db_table in self.connection.introspection.table_names(self.connection.cursor()):
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 162, in cursor
django_1 |     cursor = self.make_debug_cursor(self._cursor())
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 135, in _cursor
django_1 |     self.ensure_connection()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 130, in ensure_connection
django_1 |     self.connect()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/utils.py", line 98, in __exit__
django_1 |     six.reraise(dj_exc_type, dj_exc_value, traceback)
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 130, in ensure_connection
django_1 |     self.connect()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 119, in connect
django_1 |     self.connection = self.get_new_connection(conn_params)
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/postgresql_psycopg2/base.py", line 176, in get_new_connection
django_1 |     connection = Database.connect(**conn_params)
django_1 |   File "/usr/local/lib/python2.7/site-packages/psycopg2/__init__.py", line 164, in connect
django_1 |     conn = _connect(dsn, connection_factory=connection_factory, async=async)
django_1 | django.db.utils.OperationalError: could not connect to server: Connection refused
django_1 | 	Is the server running on host "db" (172.17.0.6) and accepting
django_1 | 	TCP/IP connections on port 5432?
django_1 | 
^C
Aborting.
**my-machine:testclone andrewya$ docker-compose stop django
Stopping testclone_django_1... done
**my-machine:testclone andrewya$ docker-compose start django
Starting testclone_django_1...
**my-machine:testclone andrewya$ docker-compose logs django
Attaching to testclone_django_1
django_1 | django-configurations version 1.0, using configuration 'Dev'
django_1 | Unhandled exception in thread started by <function wrapper at 0x7f3b6ed46b18>
django_1 | Performing system checks...
django_1 | 
django_1 | System check identified no issues (0 silenced).
django_1 | Traceback (most recent call last):
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/utils/autoreload.py", line 229, in wrapper
django_1 |     fn(*args, **kwargs)
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 116, in inner_run
django_1 |     self.check_migrations()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 168, in check_migrations
django_1 |     executor = MigrationExecutor(connections[DEFAULT_DB_ALIAS])
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/executor.py", line 19, in __init__
django_1 |     self.loader = MigrationLoader(self.connection)
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/loader.py", line 47, in __init__
django_1 |     self.build_graph()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/loader.py", line 191, in build_graph
django_1 |     self.applied_migrations = recorder.applied_migrations()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/recorder.py", line 59, in applied_migrations
django_1 |     self.ensure_schema()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/migrations/recorder.py", line 49, in ensure_schema
django_1 |     if self.Migration._meta.db_table in self.connection.introspection.table_names(self.connection.cursor()):
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 162, in cursor
django_1 |     cursor = self.make_debug_cursor(self._cursor())
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 135, in _cursor
django_1 |     self.ensure_connection()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 130, in ensure_connection
django_1 |     self.connect()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/utils.py", line 98, in __exit__
django_1 |     six.reraise(dj_exc_type, dj_exc_value, traceback)
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 130, in ensure_connection
django_1 |     self.connect()
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/base.py", line 119, in connect
django_1 |     self.connection = self.get_new_connection(conn_params)
django_1 |   File "/usr/local/lib/python2.7/site-packages/django/db/backends/postgresql_psycopg2/base.py", line 176, in get_new_connection
django_1 |     connection = Database.connect(**conn_params)
django_1 |   File "/usr/local/lib/python2.7/site-packages/psycopg2/__init__.py", line 164, in connect
django_1 |     conn = _connect(dsn, connection_factory=connection_factory, async=async)
django_1 | django.db.utils.OperationalError: could not connect to server: Connection refused
django_1 | 	Is the server running on host "db" (172.17.0.6) and accepting
django_1 | 	TCP/IP connections on port 5432?
django_1 | 
django_1 | django-configurations version 1.0, using configuration 'Dev'


^C
Aborting.
**my-machine:testclone andrewya$ docker-compose run --rm django bash
**root@45537a376999:/usr/src/app# ./manage.py test
Creating test database for alias 'default'...
Creating test database for alias 'radius'...

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
Destroying test database for alias 'default'...
Destroying test database for alias 'radius'...
**root@45537a376999:/usr/src/app# ./manage.py migrate
Operations to perform:
  Synchronize unmigrated apps: markdown, corsheaders, rest_auth, staticfiles, messages, rest_framework, django_admin_bootstrapped
  Apply all migrations: authtoken, sessions, admin, auth, radauth, contenttypes, freeradius
Synchronizing apps without migrations:
  Creating tables...
    Creating table corsheaders_corsmodel
    Running deferred SQL...
  Installing custom SQL...
Running migrations:
  Rendering model states... DONE
  Applying contenttypes.0001_initial... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0001_initial... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying radauth.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying authtoken.0001_initial... OK
  Applying freeradius.0001_initial... OK
  Applying freeradius.0002_auto_20151123_1556... OK
  Applying sessions.0001_initial... OK
**root@45537a376999:/usr/src/app# ./manage.py test -d
Creating test database for alias 'default'...
Creating test database for alias 'radius'...

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
Destroying test database for alias 'default'...
Destroying test database for alias 'radius'...
**root@45537a376999:/usr/src/app# exit
```

**Comment by DanielGabris** [comment]
@andrewyager Okay, solved my problem. I am not sure why, but it was showing the followings outpu from pip freeze after building docker:

```
root@fa5ded480a2a:/usr/src/app# pip freeze
Django==1.8.7
django-admin-bootstrapped==2.5.6
django-bootstrap3==6.2.2
django-configurations==1.0
django-cors-headers==1.1.0
django-filter==0.11.0
django-rest-auth==0.6.0
djangorestframework==3.3.1
djangorestframework-jwt==1.7.2
Markdown==2.6.4
mysqlclient==1.3.7
psycopg2==2.6.1
PyJWT==1.4.0
pyrad==2.0
python-dateutil==2.4.2
six==1.10.0
virtualenv==13.1.2
wheel==0.26.0
```
So I have installed django-radius in django bash by running this command:pip install -e 
````
git://github.com/andrewyager/django-radius.git#egg=django-radius
````
It is working now. I can get key response from login api.
Thanks

**Comment by andrewyager** [comment]
@DanielGabris I've written a test suite that now tests the django app's integrity on build, and have finally resolved issues with the CI server testing for it. Also, the changes that were need for django-radius have been pushed into the upstream, and so the commit will switch back to the pypi package for that. Make sure you update to current master shortly (once #30 has been merged, which is waiting on the CI build to finish).

**Comment by DanielGabris** [comment]
@andrewyager I was able to login as test@testing123.
How can I see stats data for this user and how can I reproduce this issue?
Do you mean the user name next to welcome message?

**Comment by andrewyager** [comment]
Most easily emulated by closing the browser and re-opening. In this case you remain logged in, but the username is lost.

**Comment by DanielGabris** [comment]
@andrewyager The username variable is being shared between two separated scopes.
I think we have to add a service like 'currentUser' to keep the username in its scope and expose to all other controllers.

**Comment by andrewyager** [comment]
Can we add it to djangoAuth and keep it there?

**Comment by DanielGabris** [comment]
Yes, we can, I will do that.

**Comment by DanielGabris** [comment]
I see it was using root scope in djangoAuth. so I think the problem will be solved by changing the $scope.username to $rootScope.username.
Let you know how it works.

**Comment by DanielGabris** [comment]
Great! It works now by using rootScope correctly.

**Comment by DanielGabris** [comment]
Hey @andrewyager I have made a pull request. If the commit is ok, could you assign a next task?
